fail2ban/server/banmanager.py

```diff
@@ -319,22 +311,39 @@ def _inBanList(self, ticket):
 	# @return the list of ticket to unban
 	
 	def unBanList(self, time):
-		try:
-			self.__lock.acquire()
+		with self.__lock:
 			# Permanent banning
 			if self.__banTime < 0:
 				return list()
 
-			# Gets the list of ticket to remove.
-			unBanList = [ticket for ticket in self.__banList if ticket.isTimedOut(time, self.__banTime)]
-			
+			# Check next unban time:
+			if self.__nextUnbanTime > time:
+				return list()
+
+			# Gets the list of ticket to remove (thereby correct next unban time).
+			unBanList = {}
+			self.__nextUnbanTime = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFL
```

Comment:
It has only once this value... But you are right, the plan was not necessarily to rewrite it after 2038, but I've confused with copy-paste from IPv6 with wide-long, indeed a smaller long would be enough. 
E. g. 0X7FFFFFFFFFFF already "Jun 20 05:22:07 GMT 4461763" (I suspect that in 4.5 Mio years we still have the same python;)

BTW. In your test above, both values are really of type long (max-int is 0x7fffffff), but of course the second one is faster, because internally can be represented in wideInt, first one - only wide arithmetic.